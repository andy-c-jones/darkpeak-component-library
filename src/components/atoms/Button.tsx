// import React from 'react';

interface ButtonProps {
  /**
   * Is this the principal call to action on the page?
   * Primary buttons should be used for the primary action on each page.
   * Secondary buttons should be used for secondary actions on each page.
   * Tertiary buttons should be used for tertiary actions on each page.
   * @default primary
   */
  type?: 'primary' | 'secondary' | 'tertiary';
  /**
   * How large should the button be?
   */
  size?: 'small' | 'medium' | 'large';
  /**
   * Button contents
   */
  label: string;
  /**
   * Optional click handler
   */
  onClick?: () => void;
  /**
   * Is the button disabled?
   */
  disabled?: boolean;
}

const sizeMap = {
  small: 'w-24 h-8',
  medium: 'w-32 h-10',
  large: 'w-40 h-12'
};

const colourMap = {
  primary: 'text-white bg-primary-default hover:bg-primary-accent',
  secondary: 'text-black bg-secondary-default hover:bg-secondary-accent',
  tertiary: 'text-white bg-tertiary-default hover:bg-tertiary-accent'
};

const disabledColourMap = {
  primary: 'text-white bg-primary-accent',
  secondary: 'text-black bg-secondary-accent',
  tertiary: 'text-white bg-tertiary-accent'
};

/**
 * Primary UI component for user interaction
 */
export const Button = ({ type = 'primary', size = 'medium', label, ...props }: ButtonProps) => {
  const colours = props.disabled ? disabledColourMap : colourMap;

  return (
    <button
      type="button"
      className={['rounded-lg', sizeMap[size], colours[type]].join(' ')}
      {...props}
    >
      {label}
    </button>
  );
};
