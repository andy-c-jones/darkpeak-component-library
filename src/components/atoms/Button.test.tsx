/**
 * @vitest-environment jsdom
 */
import { render, screen } from '@testing-library/react';
import { describe, it, expect } from 'vitest';
import { Button } from './Button';

describe('Button', () => {
  it('has a default type and therefore colour of "primary"', async () => {
    render(<Button label="Button" />);

    let button = await screen.getByRole('button');

    expect(button.className).toContain('bg-primary');
  });

  it('has a default type and therefore colour of "primary"', async () => {
    render(<Button label="Button" />);

    let button = await screen.getByRole('button');

    expect(button.className).toContain('bg-primary');
  });
});
