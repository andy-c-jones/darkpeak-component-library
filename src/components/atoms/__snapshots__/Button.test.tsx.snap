// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button > renders without error 1`] = `
<DocumentFragment>
  <button
    class="rounded-lg w-32 h-10 text-white bg-primary-default hover:bg-primary-accent"
    type="button"
  >
    test
  </button>
</DocumentFragment>
`;
